tests/core/OperationProcessor.spec.ts

```diff
@@ -459,6 +462,15 @@ describe('OperationProcessor', async () => {
         expect(newDidState!.document).toBeDefined();
         expect(newDidState!.nextRecoveryCommitmentHash).toEqual(nextRecoveryCommitmentHash);
       });
+
+      it('should apply the create operation by with { } document if encoded data and suffix data do not match', async () => {
+        isValidHashSpy.and.returnValue(false);
+        const createOperationData = await OperationGenerator.generateAnchoredCreateOperation({ transactionTime: 1, transactionNumber: 1, operationIndex: 1 });
+        const newDidState = await operationProcessor.apply(createOperationData.anchoredOperationModel, undefined);
+        expect(newDidState!.lastOperationTransactionNumber).toEqual(1);
+        expect(newDidState!.document).toEqual({ });
+        expect(newDidState!.nextRecoveryCommitmentHash).not.toEqual(nextRecoveryCommitmentHash);
```

Comment:
This check is invalid because `nextRecoveryCommitmentHash` will never be the same as `newDidState.nextRecoveryCommitmentHash`. I think it's much more direct to check the commitment hash against the double hash of the `createOperationData`'s recovery key.

Reply:
Good point